File: backend/Stockr.Domain/Entities/Purchase.cs
namespace Stockr.Domain.Entities;

public class Purchase : BaseEntity
{
    public Guid SupplierId { get; set; }
    public Supplier? Supplier { get; set; }
    public decimal TotalAmount { get; set; }
    public DateTime PurchaseDate { get; set; }
    public string? Notes { get; set; }
    public string InvoiceNumber { get; set; }
    public Guid? TenantId { get; set; }
    public Tenant? Tenant { get; set; }
    public ICollection<PurchaseItem> PurchaseItems { get; set; } = new List<PurchaseItem>();
}

File: frontend/stockr-app/src/app/models/purchase.ts
export interface Purchase {
  id: string;
  supplierId: string;
  supplierName: string;
  totalAmount: number;
  purchaseDate: string;
  notes?: string;
  invoiceNumber: string;
  purchaseItems: PurchaseItem[];
}

export interface PurchaseItem {
  id: string;
  productId: string;
  productName: string;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
}